### PR TITLE
More helpful rollup errors

### PR DESCRIPTION
**Pull request by gribnoysup**
First of all: __this is just a suggestion__, so feel free to close the PR if you are not interested 😸

This PR handles Rollup errors in a more graceful way (the change copies essential parts of [`rollup/bin/src/logging.ts`](https://github.com/rollup/rollup/blob/master/bin/src/logging.ts), unfortunately, Rollup doesn't export this part of the module to the public). Everything that could help users of the `microbundle` understand the underlying issue with the build is printed to the console.

This is how it looks for the failed build with this change:

image

I also switched microbundle from using `process.std(out|err).write` directly to using `console.(log|error)` methods. Console methods are doing the same thing, only the newlines are added automatically by console methods, so you don't need to do this whenever you are printing to the console. This also results in microbundle being a tiny bit (4 KB) smaller now, haha 🎉

### CODE REVIEW

**Comment by gribnoysup** [comment]
Seems like tests are failing for [some](https://github.com/developit/microbundle/pull/127) [time](https://github.com/developit/microbundle/pull/126) [already](https://github.com/developit/microbundle/pull/119) because the generated fixture bundles are smaller than expected 😄

Should I update the snapshots?

File: src/index.js [hunk 1]
```diff
@@ -37,17 +37,17 @@ export default async function microbundle(options) {
 		options.pkg = JSON.parse(await readFile(resolve(cwd, 'package.json'), 'utf8'));
 	}
 	catch (err) {
-		process.stderr.write(chalk.yellow(`${chalk.yellow.inverse('WARN')} no package.json found. Assuming a pkg.name of "${basename(options.cwd)}".`)+'\n');
+		stderr(chalk.yellow(`${chalk.yellow.inverse('WARN')} no package.json found. Assuming a pkg.name of "${basename(options.cwd)}".`));
 		let msg = String(err.message || err);
-		if (!msg.match(/ENOENT/)) console.warn(`  ${chalk.red.dim(msg)}`);
+		if (!msg.match(/ENOENT/)) stderr(`  ${chalk.red.dim(msg)}`);
```
Comment (gribnoysup) [inline src/index.js]: `console.warn` and `console.error` are aliases in NodeJS, this is why it is safe to replace here with `stderr` util method

File: src/utils.js [hunk 2]
```diff
@@ -6,3 +6,5 @@ export const readFile = promisify(fs.readFile);
 export const stat = promisify(fs.stat);
 export const isDir = name => stat(name).then( stats => stats.isDirectory() ).catch( () => false );
 export const isFile = name => stat(name).then( stats => stats.isFile() ).catch( () => false );
+export const stdout = console.log.bind(console); // eslint-disable-line no-console
+export const stderr = console.error.bind(console);
```
Comment (ForsakenHarmony) [inline src/utils.js]: why did you add the aliases?

**Comment by ForsakenHarmony** [comment]
the tests will be fixed when my pr is merged, so wait for that

File: src/utils.js (hunk 2 above)
Comment (ForsakenHarmony) [reply src/utils.js]: The second reason seems like a downside because you won't get that warning now with the aliases, but that was also the case with the current setup

**Comment by Andarist** [comment]
@gribnoysup could u rebase this on top of the master?

File: src/utils.js (hunk 2 above)
Comment (gribnoysup) [reply src/utils.js]: Yep, totally agreed! More or less nobody is stopping you from just using `stdout.write` if you want to avoid eslint screaming at you 😸From my point of view, this eslint rule is about _how_ you approach printing something to the terminal. Having exception just in one place doesn't suggest to other developers that are working on the project that you can just throw around `console.log` statements all over the place, compared to having `consle.log` everywhere and using `eslint-ignore`, but this is partially true for using `stdout(err)` also.

And, again, this is my highly opinionated thoughts on the topic. If you feel that this library is better off without these aliases, I don't mind removing them 😄

**Comment by gribnoysup** [comment]
@Andarist sure, give me a sec!

**Comment by Andarist** [comment]
Thanks!
